src/rdkafka_txnmgr.c

```diff
@@ -1593,20 +1653,32 @@ static void rd_kafka_txn_handle_TxnOffsetCommit(rd_kafka_t *rk,
                 int remains_ms = rd_timeout_remains(rko->rko_u.txn.abs_timeout);
 
                 if (!rd_timeout_expired(remains_ms)) {
-                        rd_kafka_coord_req(
-                            rk, RD_KAFKA_COORD_GROUP,
-                            rko->rko_u.txn.cgmetadata->group_id,
-                            rd_kafka_txn_send_TxnOffsetCommitRequest, rko,
-                            rd_timeout_remains_limit0(
-                                remains_ms, rk->rk_conf.socket_timeout_ms),
-                            RD_KAFKA_REPLYQ(rk->rk_ops, 0),
-                            rd_kafka_txn_handle_TxnOffsetCommit, rko);
+                        rd_ts_t diff =
+                            rd_interval(&last_retry, 1000 * 1000 /* 1s */, 0);
+                        if (diff < 0) {
+                                rd_kafka_dbg(
+                                    rk, EOS, "TXNOFFSETCOMMIT",
+                                    "Waiting %ld µs before next retry. rkb:%s "
```

Comment:
i think there's a macro you should use rather than %ld, look at other examples.
and do you really want to report in micro-seconds? i think better to use milliseconds. or if that's too coarse, no log message at all.